server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java

```diff
@@ -361,7 +362,7 @@ private CompletionStage<RestResponse> streamKeys(RestRequest request) {
       int batch = batchParam == null || batchParam.isEmpty() ? STREAM_BATCH_SIZE : Integer.parseInt(batchParam);
       int limit = limitParam == null || limitParam.isEmpty() ? -1 : Integer.parseInt(limitParam);
 
-      Cache<?, ?> cache = invocationHelper.getRestCacheManager().getCache(cacheName, TEXT_PLAIN, TEXT_PLAIN, request);
+      Cache<?, ?> cache = invocationHelper.getRestCacheManager().getCache(cacheName, TEXT_PLAIN, MATCH_ALL, request);
```

Comment:
So the test still passes if I revert this. Maybe something missing?

Reply:
Yeah, with the `TEXT_PLAIN` matcher it works as expected, the only operation we do with the returned cache is over the keys. I've changed this mostly to be clear that values with any type are ok. I can revert it.